packages/client-web/__tests__/integration/web_client.test.ts

```diff
@@ -16,46 +16,85 @@ describe('[Web] Client', () => {
           'Test-Header': 'foobar',
         },
       })
-      const fetchParams = await pingAndGetRequestInit(client)
+      await client.ping()
+      const fetchParams = getFetchRequestInit()
       expect(fetchParams!.headers).toEqual({
-        Authorization: 'Basic ZGVmYXVsdDo=', // default user with empty password
+        ...defaultHeaders,
         'Test-Header': 'foobar',
       })
     })
 
     it('should work with no additional HTTP headers provided', async () => {
       const client = createClient({})
-      const fetchParams = await pingAndGetRequestInit(client)
-      expect(fetchParams!.headers).toEqual({
-        Authorization: 'Basic ZGVmYXVsdDo=', // default user with empty password
+      await client.ping()
+      const fetchParams = getFetchRequestInit()
+      expect(fetchParams!.headers).toEqual(defaultHeaders)
+    })
+
+    it('should work with additional HTTP headers on the method level', async () => {
+      const client = createClient({
+        http_headers: {
+          FromInstance: 'foo',
+        },
       })
+
+      let fetchCalls = 1
+      const testParams = getHeadersTestParams(client)
+      for (const param of testParams) {
+        await param.methodCall({ FromMethod: 'bar' })
+        expect(getFetchRequestInit(fetchCalls++).headers)
+          .withContext(
+            `${param.methodName}: merges custom HTTP headers from both method and instance`,
+          )
+          .toEqual({
+            ...defaultHeaders,
```

Comment:
can we add a test that the Authorization header is not overridden when provided in `http_headers`?